packages/sanity/src/structure/panes/documentList/sheetList/SheetListCell.tsx

```diff
@@ -92,7 +97,17 @@ export function SheetListCell(cell: Cell<DocumentSheetTableRow, unknown>) {
     setSelectedAnchorCell(column.id, row.index)
   }, [column.id, row.index, setSelectedAnchorCell])
   const cellState = getStateByCellId(cell.column.id, cell.row.index)
-  const {patchDocument, unsetDocumentValue} = cellContext.table.options.meta || {}
+  const {patchDocument, unsetDocumentValue} =
+    (!fieldType?.readOnly && cellContext.table.options.meta) || {}
```

Comment:
Gotcha, but I think it will be not easy to understand in the future and if later we want to add something else to meta which we want to access it won't be possible because of this check.
What do you think if we move this to the  `handlePatchField` and  `handleUnsetField` functions?

Reply:
You'd like this only for the patch and unset? I ask because we also need to prevent the updating of the cell state [on line 105 too](https://github.com/sanity-io/sanity/pull/6954/files#diff-ed541947aa0eaaf930a785ff2ff9a1eb96240207c108c889e391d3a7a14034e2R105)